[WIP]I589 Update Workspace ->Volume Mount (#595)
codenvy/che-docs: src/main/pages/workspace-admin/volumes.adoc

---

[id="default-volumes_for_workspace_containers"]
== Default volumes for workspace containers

> By default, workspace containers start with a default volume and have a minimum of one PVC that is located in `/projects` directory. 

Reviewer: ```suggestion
By default, workspace containers start with a default volume and have a minimum of one PVC that is located in the `/projects` directory. 
```

Workspace projects are physically located in the `/projects` directory. When a workspace stops, the machines are destroyed, but the volumes persist.

[id="Adding-volumes"]
== Adding volumes

> Your workspace may need additional volumes if your data must persist for a local Maven repository, node_modules folder, ruby jems, or authorized_keys file for SSH connections. You can add additional volumes for your workspace machines. Each machine can add as many volumes as the underlying infrastructure can support. OpenShift may impose a limit on the number of volumes.

Reviewer 2: Typos/need styling: `node_modules` directory (not folder), Ruby gems, `authorized_keys` file.

Reviewer: + no 'if'

```suggestion
In order for your data to persist for a local Maven repository, the `node_modules/` directory, Ruby gems, or the `authorized_keys` file for SSH connections, your workspace will need additional volumes. Each machine can add as many volumes as the underlying infrastructure can support. OpenShift may impose a limit on the number of volumes.
```

}
----

To avoid failures when updating the workspace configuration using REST APIs: 

> * Be sure to use an absolute path. 

Reviewer: ```suggestion
* Use an absolute path. 
```

> * The name and path cannot contains special characters, including `-` and `_`.

Reviewer 2: Typo: "contains" should be "contain"

"including `-` or `_`"

> * The name and path cannot contains special characters, including `-` and `_`.

Reviewer: ```suggestion
* The name and path cannot contain special characters, including dashes (`-`) or underscores (`_`).
```

[NOTE]
====

> If you want your workspace machines to share volumes, create volumes for each machine that has an identical name. This will allow machines to share the same volume.

Reviewer: no 'if'

```suggestion
To allow machines to share the same volume, create a volume for each machine with an identical name. 
```
